doc/source/whatsnew/v1.2.0.rst

```diff
@@ -132,6 +132,7 @@ Plotting
 Groupby/resample/rolling
 ^^^^^^^^^^^^^^^^^^^^^^^^
 
+- Bug in :meth:`DataFrameGroupBy.apply` that would some times throw an erroneous ``ValueError: cannot reindex from a duplicate axis`` (:issue:`16646`)
```

Comment:
Can you add some context as to when this might happen? I think axis=1 and MultiIndex columns, right?

Reply:
Done.